src/main/java/stirling/software/SPDF/controller/api/MergeController.java

```diff
@@ -53,90 +53,97 @@ private Comparator<MultipartFile> getSortComparator(String sortType) {
     @Operation(
             summary = "Merge multiple PDF files into one",
             description =
                     "This endpoint merges multiple PDF files into a single PDF file. The merged file will contain all pages from the input files in the order they were provided. Input:PDF Output:PDF Type:MISO")
     public ResponseEntity<byte[]> mergePdfs(@ModelAttribute MergePdfsRequest form)
             throws IOException {
-        List<File> filesToDelete = new ArrayList<File>();
+        List<File> tempFiles = new ArrayList<>();
         try {
             MultipartFile[] files = form.getFileInput();
             Arrays.sort(files, getSortComparator(form.getSortType()));
 
-            PDFMergerUtility mergedDoc = new PDFMergerUtility();
-            ByteArrayOutputStream docOutputstream = new ByteArrayOutputStream();
+            PDFMergerUtility pdfMerger = new PDFMergerUtility();
+            ByteArrayOutputStream mergedPdfStream = new ByteArrayOutputStream();
 
             for (MultipartFile multipartFile : files) {
-                File tempFile = GeneralUtils.convertMultipartFileToFile(multipartFile);
-                filesToDelete.add(tempFile);
-                mergedDoc.addSource(tempFile);
+                File tempFile = createTempFile(multipartFile);
+                tempFiles.add(tempFile);
+                pdfMerger.addSource(tempFile);
             }
 
-            mergedDoc.setDestinationFileName(
-                    files[0].getOriginalFilename().replaceFirst("[.][^.]+$", "") + "_merged.pdf");
-            mergedDoc.setDestinationStream(docOutputstream);
-
-            mergedDoc.mergeDocuments(null);
+            String mergedFileName = getMergedFileName(files);
+            pdfMerger.setDestinationFileName(mergedFileName);
+            pdfMerger.setDestinationStream(mergedPdfStream);
+            pdfMerger.mergeDocuments(null);
 
             return WebResponseUtils.bytesToWebResponse(
-                    docOutputstream.toByteArray(), mergedDoc.getDestinationFileName());
+                    mergedPdfStream.toByteArray(), mergedFileName);
         } catch (Exception ex) {
-            logger.error("Error in merge pdf process", ex);
+            logger.error("Error occurred during PDF merge", ex);
             throw ex;
         } finally {
-            for (File file : filesToDelete) {
-                file.delete();
-            }
+            cleanupTempFiles(tempFiles);
+        }
+    }
+
+    private File createTempFile(MultipartFile multipartFile) throws IOException {
+        File tempFile = GeneralUtils.convertMultipartFileToFile(multipartFile);
+        tempFile.deleteOnExit();
+        return tempFile;
+    }
+
+    private String getMergedFileName(MultipartFile[] files) {
+        return files[0].getOriginalFilename().replaceFirst("[.][^.]+$", "") + "_merged.pdf";
+    }
+
+    private void cleanupTempFiles(List<File> tempFiles) {
+        for (File file : tempFiles) {
+            file.delete();
```

Comment:
I feel like these are all operations that could fit within the Util class and used elsewhere if needed?